### PR TITLE
test: compressed bundle size is less than 1MB

### LINKED ISSUE #36: Test that compressed contract bundles are less than 1MB
The chain sdk has limitations on a tx message size - approx 1MB. See: https://github.com/Agoric/agoric-sdk/issues/7501. 

We should add a test that ensures compressed versions of the generated bundles do not exceed 1MB.  Suggested: https://github.com/Agoric/dapp-offer-up/pull/34#discussion_r1441081251

See here for an example of how to compress a bundle: https://github.com/DCFoundation/cosmos-proposal-builder/blob/84318e05322ed46653dbdd6634d0f9f1f304e180/src/lib/compression.ts#L27-L38

And here for a filepath for generated bundles from ci: https://github.com/Agoric/dapp-offer-up/actions/runs/7403614239/job/20143726617#step:6:24

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
closes #36 

```bash
✔ build-proposal › proposal builder generates compressed bundles less than 1MB (1.9s)
  ℹ agoric run stdout: agoric: run: Deploy script will run with Node.js ESM
    bundle-source --to /Users/0xpatrick/repos/dapp-offer-up/contract/bundles /Users/0xpatrick/repos/dapp-offer-up/contract/src/offer-up.contract.js offer-up
    creating start-offer-up-permit.json
    creating start-offer-up.js
    You can now run a governance submission command like:
      agd tx gov submit-proposal swingset-core-eval start-offer-up-permit.json start-offer-up.js \
        --title="Enable <something>" --description="Evaluate start-offer-up.js" --deposit=1000000ubld \
        --gas=auto --gas-adjustment=1.2
    Remember to install bundles before submitting the proposal:
      agd tx swingset install-bundle @/Users/0xpatrick/.agoric/cache/b1-88b354fd27623534134d369f89d20bbb7ed82420bba99669eba34a8b1847345ff43418e0c8cf4dac1ed4e8102eca534f5ba7c4e695239f01f074ad84f62cfa74.json
      agd tx swingset install-bundle @/Users/0xpatrick/.agoric/cache/b1-5a4ee4b4f5106eca1ba71fd7f8cf97875f162c2439ca1077375e2adf4a5fe29c4f5b811b1bbc4040e5f45ca617e141f7dc64374660d8945eb8783c7bb8edc22f.json

  ℹ {
      bundleId: 'b1-88b354fd27623534134d369f89d20bbb7ed82420bba99669eba34a8b1847345ff43418e0c8cf4dac1ed4e8102eca534f5ba7c4e695239f01f074ad84f62cfa74',
      compressedSize: '0.3141002655029297 MB',
    }
  ℹ {
      bundleId: 'b1-5a4ee4b4f5106eca1ba71fd7f8cf97875f162c2439ca1077375e2adf4a5fe29c4f5b811b1bbc4040e5f45ca617e141f7dc64374660d8945eb8783c7bb8edc22f',
      compressedSize: '0.1577749252319336 MB',
    }
 ```
 
 
 
 ### Questions
 
1.  I was going to put this in `contract/test/test-bundle-source.js`, but got confused as that seems to only produces one bundle.  Should this test be added there or is this separate file fine?

2. ~~I'm not sure this will work in CI - `../bundles/bundle-offer-up.js` is present in my local but not sure it will be at this point in CI.~~ seems to be fine: https://github.com/Agoric/dapp-offer-up/actions/runs/7647247910/job/20837765045?pr=56#step:7:26

### CODE REVIEW

File: contract/test/test-build-proposal.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { Buffer } from 'buffer';
+
+const compressFile = async filePath => {
+  const fileContents = await fs.readFile(filePath, 'utf8');
+  const buffer = Buffer.from(fileContents, 'utf-8');
+  const compressed = await promisify(gzip)(buffer);
+  return compressed;
+};
+
+test('proposal builder generates bundles less than 5MB', async t => {
```
Comment (0xpatrickdev) [inline contract/test/test-build-proposal.js:16]:
```suggestion
test('proposal builder generates compressed bundles less than 1MB', async t => {
```

File: contract/test/test-build-proposal.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable import/order -- https://github.com/endojs/endo/issues/1235 */
+import { test } from './prepare-test-env-ava.js';
+import { execa } from 'execa';
+import { promises as fs } from 'fs';
+import { makeCompressFile } from './utils.js';
+
+test.before(t => (t.context.compressFile = makeCompressFile(fs.readFile)));
+
+test('proposal builder generates compressed bundles less than 1MB', async t => {
+  const { stdout } = await execa('agoric', [
```
Comment (dckc) [inline contract/test/test-build-proposal.js:10]: This doesn't seem to make use of any `execa` feature that isn't provided by node's `child_process`. Am I missing something?

**Review by dckc — APPROVED** [review]
good to have this

I don't understand the motivation to add a dependency on execa, though

File: contract/test/test-build-proposal.js (hunk 2 above)
Comment (0xpatrickdev) [reply contract/test/test-build-proposal.js]: `execa` dep removed

**Pull request merged**